plutus-chain-index/src/Plutus/ChainIndex/SyncStats.hs

```diff
@@ -79,31 +76,15 @@ instance Pretty SyncState where
     Syncing pct -> "Syncing (" <> pretty (printf "%.2f" pct :: String) <> "%)."
     NotSyncing  -> "Not syncing."
 
--- | Read 'ChainSyncEvent's for the 'TChan' every 30 seconds and log syncing summary.
-logProgress :: forall effs.
-    ( Member (LogMsg SyncLog) effs
-    , LastMember IO effs
-    )
-    => TChan SyncStats
-    -> Eff effs ()
-logProgress broadcastChan = do
-    chan <- liftIO $ atomically $ dupTChan broadcastChan
-    go chan 30_000_000 -- 30s
-  where
-    go chan delay = do
-        liftIO $ threadDelay (fromIntegral delay)
-        syncStats <- liftIO $ foldTChanUntilEmpty chan
-        let syncState = getSyncStateFromStats syncStats
-        case syncState of
-          NotSyncing -> do
-              logWarn $ SyncLog syncState syncStats (fromMicroseconds delay)
-              go chan 300_000_000 -- 300s
-          Synced -> do
-              logInfo $ SyncLog syncState syncStats (fromMicroseconds delay)
-              go chan 300_000_000 -- 300s
-          Syncing _ -> do
-              logInfo $ SyncLog syncState syncStats (fromMicroseconds delay)
-              go chan 30_000_000 -- 30s
+-- | Log syncing summary.
+logProgress :: forall effs. (Member (LogMsg SyncLog) effs) => [ChainSyncEvent] -> TimeSpec -> Eff effs ()
+logProgress events period = do
+    let syncStats = foldl (<>) mempty $ map convertEventToSyncStats events
+    let syncState = getSyncStateFromStats syncStats
+    let syncLog = SyncLog syncState syncStats (fromMicroseconds $ toNanoSecs period `div` 1000)
+    case syncState of
+      NotSyncing -> pure () -- logWarn syncLog
```

Comment:
To turn off `Not syncing. Applied 0 blocks, 0 rollbacks.` message at the beginning.

Reply:
Keep this log message. However, you need to change the implementation of `processEventsQueue`. The reason it applied 0 blocks in the beginning is because you suppose that the first event is either Resume or RollBackwards, and the rest are RollForward. However, the two first events are actually "Resume" AND "RollBackwards", so there are 2 events to skip before the RollForward. Therefore, you need to make the implementation a bit more general.

Reply:
Yeah, I know, I turned it off to avoid changing stuff as it gets expensive to change logic just to support one logging line.

Will fix it.